File: Dieries1_FrontEnd/WebContent/controller/FriendController.js
/**
 * 
 */
app.controller('FriendController',function(FriendService,$location,$scope,$rootScope)
		{
			//GET ALL USERS
			//$scope.getAllUsers = function()
			function getAllUsers()
			{
				console.log("inside getAllUsers");
				FriendService.getAllUsers().then(function(response){
					$scope.users = response.data;
					console.log(response.data)
					//$location.path('/getAllUsers');
				},
				function(response)
				{
					console.log(response.status);
					if(response.status == 401)
						{
							$location.path('/login');
						}
				})
			}
				
				//GET ALL FRIENDS
				function listOfFriends()
				{
					console.log('In get Friends')
					FriendService.listOfFriends().then(function(response)
					{
						$scope.friends = response.data //List<Friend> select * from friend where status='A' and (fromId=? or toId=?)
						$rootScope.noOfFriends=$scope.friends.length
					},
					function(response){
						if(response.status==401)
							$location.path('/login')
					})
				}
				
				//LIST OF PENDING REQUESTS
				function pendingRequests()
				{
					console.log('in pending Requests');
					FriendService.pendingRequests().then(function(response){
						$rootScope.pendingRequests = response.data;
						console.log(response.data.length)
						//$location.path('/pendingRequests');
					},
					function(response){
						if (response.status == 401)
							{
							$location.path('/login')
							}
					})
				}
				
				//GET ALL USERS
				$scope.getUserByUsername = function(fromId)
				{
					console.log('in getUserBy userName');
					FriendService.getUserByUsername(fromId).then(function(response)
							{
						console.log('in getUserBy userName 2');
								console.log(fromId);
								$scope.showUserDetails = true;
								console.log($scope.showUserDetails);
								$scope.user = response.data;
								console.log(response.data);
							},
							function(response)
							{
								console.log(response.data);
							})
				}
				
				
				//UPDATING PENDING FRIEND REQUEST
				$scope.updatePendingRequest = function(request,value){
					console.log('pending request ' + request)
					request.status=value //value is 'A' for accept and 'D' for delete
					console.log('after assigning value to status  ' + request)
					FriendService.updatePendingRequest(request).then(function(response){
						pendingRequests();
						$location.path('/pendingrequests')
					},function(response){
						if(response.status==401)
							$location.path('/login')
					})
				}
			
			
			//TO SEND A FRIEND REQUEST
			$scope.sendFriendRequest = function(toId)
			{
				FriendService.sendFriendRequest(toId).then(function(response)
						{
							console.log('in sendFriendRequest');
							console.log(toId);
							$scope.getUserByUsername(toId);
							getAllUsers();
							
							$location.path('/getAllUsers');
						},
						function(response)
						{
							console.log(response.data);
						})
			}
			
			
			pendingRequests();
			listOfFriends();
			getAllUsers();
		})

File: Diaries1/src/main/java/com/controller/FriendController.java
package com.controller;

import java.util.List;

import javax.servlet.http.HttpSession;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

import com.dao.FriendDao;
import com.dao.UserDao;
import com.model.Error;
import com.model.Friend;
import com.model.User;

@Controller
public class FriendController 
{
	@Autowired
	FriendDao friendDao;
	@Autowired
	UserDao userDao;
	
	//GET THE LIST OF USERS
	@RequestMapping("/getAllUsers")
	public ResponseEntity<?> getAllUsers(HttpSession session)
	{
		System.out.println("in friend controller");
		String username = (String) session.getAttribute("username");
		if(username == null)
		{
			Error error = new Error(5, "UnAuthorized User");
			return new ResponseEntity<Error>(error,HttpStatus.UNAUTHORIZED);
		}
		
		List<User> users = (List<User>)friendDao.getAllUsers(username);
		return new ResponseEntity<List<User>>(users,HttpStatus.OK);
	}
	
	//SEND FRIEND REQUEST
	@RequestMapping(value="/sendFriendRequest/{toId}", method = RequestMethod.POST)
	public ResponseEntity<?> sendFriendRequest(@PathVariable String toId,HttpSession session)
	{
		String username = (String) session.getAttribute("username");
		if(username == null)
		{
			Error error = new Error(5,"UnAuthorized User");
			return new ResponseEntity<Error>(error,HttpStatus.UNAUTHORIZED);
		}
		System.out.println(username + " send friend request to " + toId);
		Friend friend = new Friend();
		friend.setFromId(username);
		friend.setToId(toId);
		friend.setStatus('P');
		friendDao.friendRequest(friend);
		
		return new ResponseEntity<Friend>(friend,HttpStatus.OK);
	}
	
	//SEND LIST OF PENDING REQUESTS
	@RequestMapping(value="/pendingRequests")
	public ResponseEntity<?> pendingFriendRequests(HttpSession session)
	{
		String username = (String)session.getAttribute("username");
		
		if(username == null)
		{
			Error error = new Error(5,"UnAuthorized User");
			return new ResponseEntity<Error>(error,HttpStatus.UNAUTHORIZED);
		}
		
		List<Friend> pendingRequests = friendDao.pendingRequests(username);
		return new ResponseEntity<List<Friend>>(pendingRequests,HttpStatus.OK);
	}
	
	//GET USER BY USERiD
	@RequestMapping("/getUserByUsername/{fromId}")
	public ResponseEntity<?> getuserByUsername(@PathVariable String fromId,HttpSession session)
	{
		if(session.getAttribute("username") == null)
		{
			Error error = new Error(5 ,"Unauthorized User");
			return new ResponseEntity<Error>(error,HttpStatus.UNAUTHORIZED);
		}
		
		User user = userDao.getUserByUsername(fromId);
		return new ResponseEntity<User>(user,HttpStatus.OK);
	}
	
	//UPDATE FRIEND REQUEST EITHER ACCEPT OR REJECT
	@RequestMapping(value="/updateFriendRequest", method=RequestMethod.PUT)
	public ResponseEntity<?> updateFriendRequest(@RequestBody Friend friend,HttpSession session)
	{
		if(session.getAttribute("username") == null)
		{
			Error error = new Error(5,"unAuthorized User");
			return new ResponseEntity<Error>(error,HttpStatus.UNAUTHORIZED);
		}
		
		friendDao.updatePendingRequest(friend);
		return new ResponseEntity<Friend>(friend,HttpStatus.OK);
	}
	
	//TO GET THE LIST OF FRIENDS
	@RequestMapping("/listOfFriends")
	public ResponseEntity<?> listOfFriends(HttpSession session)
	{
		String username = (String) session.getAttribute("username");
		if(username == null)
		{
			Error error = new Error(5,"UNAUTHORIZED USER");
			return new ResponseEntity<Error>(error,HttpStatus.UNAUTHORIZED);
		}
		List<Friend> friends = friendDao.listOfFriends(username);
		return new ResponseEntity<List<Friend>>(friends,HttpStatus.OK);
	}
}
